src/net/sf/launch4j/config/Config.java

```diff
@@ -356,6 +357,16 @@ public boolean isRestartOnCrash() {
 	public void setRestartOnCrash(boolean restartOnCrash) {
 		this.restartOnCrash = restartOnCrash;
 	}
+
+	/** Restart the application whenever the java application returns the status given here **/
+	public Integer getRestartOnStatus() {
+		return restartOnStatus;
+	}
+
+	public void setRestartOnStatus(Integer status) {
+		if(status < 5 && status > 255) throw new NumberFormatException("Restart-Status needs to be between 5 and 255!");
```

Comment:
not sure if there isn't a better place to check this. I assumed 5 is out of range of the predefined 0,1,2